tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -856,11 +898,30 @@ def test_split_observation_pairs() -> None:
             (-float("inf"), [-5.0]),  # COMPLETE
         ],
         2,
+        None,
     )
     assert list(indices_below) == [0, 3]
     assert list(indices_above) == [1, 2]
 
 
+def test_split_observation_pairs_with_constraints() -> None:
```

Comment:
Why not test both cases, ``if n_below >= len(idx) or violation_1d[idx[n_below]] > 0` or not then?